### ISSUE TITLE
Legacy references to Message and MessageValidator classes in SDK documentation

### ISSUE THREAD

**Issue by j13k** [aws/aws-sdk-php#2027]
As of **May 2020**, the SDK documentation still indicates that the `Message` and `MessageValidator` classes are available in the main library, when according to issue #625 these classes were migrated to an auxilliary library in 2015.
 
https://docs.aws.amazon.com/aws-sdk-php/v3/api/namespace-Aws.Sns.html

![Namespace_Aws_Sns___AWS_SDK_for_PHP_3_x](https://user-images.githubusercontent.com/1084378/83222077-ca67a300-a1ba-11ea-8b18-696ed15547fa.png)

I'm presuming the documentation is auto-generated, but it's not clear when it was last updated.

For what it's worth, the main SNS docs also indicate this:

https://docs.aws.amazon.com/sns/latest/dg/sns-verify-signature-of-message.html

![image](https://user-images.githubusercontent.com/1084378/83222453-b40e1700-a1bb-11ea-946e-2d1ac4221715.png)

**Comment by diehlaws** [comment]
Hi @j13k, thanks for bringing this to our attention. Message validation for SNS has indeed been moved to [a separate package](https://github.com/aws/aws-php-sns-message-validator/), we'll work on updating our documentation to reflect this change. While we cannot make changes to the SNS developer guide, I'll be reaching out to their docs team internally to have the page you mention updated to highlight the requirement of this package for message validation using the AWS SDK for PHP.

**Comment by j13k** [comment]
Thanks for responding so promptly @diehlaws. I did discover that information about the separate package, but only after stumbling upon the discussion in issue #625 via a general web search.

As a more general comment, it would be great if the documentation could be regenerated/updated whenever a new release of the SDK is minted. Clearly these SNS message classes haven't been included in the main SDK for quite some time—to be honest it's a 'janky' developer experience when the docs aren't in parity with the code they're describing.

**Comment by jmcgrory** [comment]
I agree with @j13k here, the [documentation](https://docs.aws.amazon.com/sns/latest/dg/sns-verify-signature-of-message.html) is possibly intentionally vague to allow for changes in the API?  

> When possible, use one of the supported AWS SDKs for Amazon SNS to validate and verify messages. For example, with the AWS SDK for PHP you would use the `isValid` method from the `MessageValidator` class.

From the above it seems that this would be available in "**the** AWS SDK for PHP" (or even in multiple "use one of"). However it appears to now be in a separate/dedicated ancilliary package.

**Comment by ameotoko** [comment]
No progress😁

**Comment by j13k** [comment]
> We have noticed this issue has not recieved attention in 1 year. We will close this issue for now. If you think this is in error, please feel free to comment and reopen the issue.

The published documentation still seems to be out-of-date.

**Comment by stobrien89** [comment]
Hi all,

Apologies for the very delayed response here.  The docs aren't necessarily out-of-date— we intentionally pull this in from the other package and include it in our API reference because we do not have an api reference for our other packages.  We don't plan to add separate API references for our other packages either.

I've opened #3193 to add a notice that `MessageValidator` is maintained in a separate package, with a link to the SNS Message Validator package.
